### ISSUE TITLE
Invalid BAM lines

### ISSUE THREAD

**Issue by cjfields** [barricklab/breseq#146]
I was attempting to add read groups to the BAM files and ran into this using Picard AddOrReplaceReadGroups:

```
[cjfields@compute-0-19 2018-01-19-realn-test]$ java -jar $EBROOTPICARD/picard.jar AddOrReplaceReadGroups       I=genomic_bams/R1823MIC.bam       O=genomic_bams/R1823MIC.rg.bam       RGID=R1823MIC       RGLB=1       RGPL=illumina       RGPU=unit1       RGSM=R1823MIC
[Fri Jan 19 15:38:46 CST 2018] picard.sam.AddOrReplaceReadGroups INPUT=genomic_bams/R1823MIC.bam OUTPUT=genomic_bams/R1823MIC.rg.bam RGID=R1823MIC RGLB=1 RGPL=illumina RGPU=unit1 RGSM=R1823MIC    VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
[Fri Jan 19 15:38:46 CST 2018] Executing as cjfields@compute-0-19 on Linux 3.10.0-693.11.1.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16; Picard version: 2.9.0-1-gf5b9f50-SNAPSHOT
INFO	2018-01-19 15:38:46	AddOrReplaceReadGroups	Created read group ID=R1823MIC PL=illumina LB=1 SM=R1823MIC

[Fri Jan 19 15:38:50 CST 2018] picard.sam.AddOrReplaceReadGroups done. Elapsed time: 0.07 minutes.
Runtime.totalMemory()=2058354688
To get help, see http://broadinstitute.github.io/picard/index.html#GettingHelp
Exception in thread "main" htsjdk.samtools.SAMFormatException: SAM validation error: ERROR: Read name 1:259041-M2, No real operator (M|I|D|N) in CIGAR
	at htsjdk.samtools.SAMUtils.processValidationErrors(SAMUtils.java:448)
	at htsjdk.samtools.BAMRecord.getCigar(BAMRecord.java:252)
	at htsjdk.samtools.SAMRecord.getAlignmentEnd(SAMRecord.java:603)
	at htsjdk.samtools.SAMRecord.computeIndexingBin(SAMRecord.java:1547)
	at htsjdk.samtools.SAMRecord.isValid(SAMRecord.java:2054)
	at htsjdk.samtools.BAMFileReader$BAMFileIterator.advance(BAMFileReader.java:795)
	at htsjdk.samtools.BAMFileReader$BAMFileIterator.next(BAMFileReader.java:781)
	at htsjdk.samtools.BAMFileReader$BAMFileIterator.next(BAMFileReader.java:751)
	at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:569)
	at htsjdk.samtools.SamReader$AssertingIterator.next(SamReader.java:543)
	at picard.sam.AddOrReplaceReadGroups.doWork(AddOrReplaceReadGroups.java:140)
	at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:205)
	at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:94)
	at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:104)
```

After a bit of searching on that error, I found that the BAM output is generating a LOT of reads that are showing as entirely soft-clipped (I'm guessing these should be unmapped)?  

```
[cjfields@compute-0-19 2018-01-19-realn-test]$ samtools view genomic_bams/R1823MIC.bam | awk -F '\t' '($6 ~ /^[0-9]+S$/)' | head
1:259041-M2	0	NZ_JRYM01000009	92219	255	249S	*	0	0	GGATAAACCGGCAAAAACCGGGCAATCGCAAAAAGGCGTAATGTGTTCCCCGCGCCAGCGGGGATAAACCTGTGTTTGCGGCATTAACGCTCACCAGCATTTCGTGTTCCCCGCGCCAGCGGGGATAAACCGACGTGGTCATGGGTGCTGCTGTTGCAGAGCCAGTGTTCCCCGCGCCAGCGGGGATAAACCGAGCAGATACACGGCTTTGTATTCCGTGCGCCCGTGTTCCCCGCGCCAGCGGGGATA	DDDDDIIIIIIIIIIIIIIIIIHIHHIIIIIIIHEHHHHIIIHIIIIIIIIIIIIIIIIIIIHHHIIIIIIIIIIIIIIGIIIGHIIIIIIIIIIIIHIIIFEHHIIIIHIIIIIIHIIIIHIHDHHHHIIIGIHIIIIHIIHIIIIHIIHHFHIIIIIIHIIIIIHIIGIIIIHDHIHHHIHDHIGIIIIIIIHDHIIIIIHIGHIIHIGIFHHIIIIGICHEHHHHHIHHFHIIIIIGHIIIIHIHI	AS:i:249	X1:i:1	X2:i:0	XJ:i:2	XL:i:0	XR:i:1
1:289152-M2	0	NZ_JRYM01000009	92219	255	250S	*	0	0	TAGAGTGTTCCCCGCGCCAGCGGGGATAAACCGGCAAAAACCGGGCAATCGCAAAAAGGCGTAATGTGTTCCCCGCGCCAGCGGGGATAAACCTGTGTTTGCGGCATTAACGCTCAGCATTTCGTGTTCCCCGCGCCAGCGGGGATAAACCGACGTGGTCATGGGTGCTGCTGTTGCAGAGCCAGTGTTCCCCGCGCCAGCGGGGATAAACCGAGCAGATACACGGCTTTGTATTCCGTGCGCCCGTGTT	DDDDDFHIIIIIIIIIIIIGIIIIIHIIIIIIIIIIHIIIIHIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIHHIIIIIIIIIICHIIIIIIIIIHIIIIIGHHIIIIIIIIIIIIIIIIIIIIHIIIHHIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIHIIIIIHIIIIIIIGIIIIIIIIIIIIIIIIIIIIIIIIIIHHHHIIIGIIIIHIIHIIIIGIIIIIIEHHIGIHIIIIHHHHIH	AS:i:239	X1:i:1	X2:i:0	XJ:i:2	XL:i:0	XR:i:2
1:319271-M1	0	NZ_JRYM01000009	92219	255	249S	*	0	0	TGTTCCCCGCGCCAGCGGGGATAAACCGAATAGCAATAGTCCATAGATTTGCGAAAACAGGTGTTCCCCGCGCCAGCGGGGATAAACCGGAGCCTGACGAGACTACTGAGGCCGTTCTGTCGAGTTCCCCGCGCCAGCGGGGATAAACCACCATATAACCCGTTATCTCTTTCTCAAGTTTTTATATTAGCAGTACTTGTAATAAGCAACATATCCACGTAACACCTCATGTTCAAAATAGTTCTCCAT	DDDDDIIIIIIIIIIIIIIGHIIIIIIIHHHIIHGHFHIIHGIIGIGHIIIIIIHHIIHIIIIIIIHHHIHIIGHIHIIGIDHHIIIGIHHIHHHIIIFCEEHHIGHIIIHIIIHGHHIIIIHIIIIGIIIGIIIHHGIIHIHIHHGIIHHHEHHHHGHHHIIIIHIHFCHHEHIIIIIIIIIHHHIIHIIFHIGHFHIHHHIIFHIHEHHIIGG?GHHICHIICHIHHHHHIIII?BF?F@@A@FHHF	AS:i:243	X1:i:1	X2:i:0	XJ:i:0	XL:i:3	XR:i:0
1:473479-M2	0	NZ_JRYM01000009	92219	255	250S	*	0	0	TGTTCCCCGCGCCAGCGGGGATAAACCGGCAAAAACCGGGCAATCGCAAAAAGGCGTAATGTGTTCCCCGCGCCAGCGGGGATAAACCTGTGTTTGCGGCATTAACGCTCACCAGCATTTCGTGTTCCCCGCGCCAGCGGGGATAAACCGACGTGGTCATGGGTGCTGCTGTTGCAGAGCCAGTGTTCCCCGCGCCAGCGGGGATAAACCGAGCAGATACACGGCTTTGTATTCCGTGCGCCCGTGTTCC	DDDDDIFHIIIIIIHIHIIIDHIIIIIIHIIIIIIIHIIIIIIIIHHHIHHIIIIIIIIIIIIIIIIIIIHIIIIIIIIIIIIIIIIIIHHIIHHIIIHIHDGHHIIIIIIIIEHHIIIHIIIIIIIIGIIIHIIIIIGIIIIHFHHHF<FHIIHIIIIIHHHIIIIGHIIIIHIIIIHHHIHHHIIIHHIIHIIIHHIIIDHIIIIIHHHHHIIIIIIIIIGIIIIIHH?HH?BAGBEH<HEHEEHCFH	AS:i:250	X1:i:1	X2:i:0	XJ:i:2	XL:i:0	XR:i:2
1:478990-M1	0	NZ_JRYM01000009	92219	255	248S	*	0	0	CCGCGCCAGCGGGGATAAACCGAATAGCAATAGTCCATAGATTTGCGAAAACAGGTGTTCCCCGCGCCAGCGGGGATAAACCGGAGCCTGACGAGACTACTGAGGCCGTTCTGTCGAGTTCCCCGCGCCAGCGGGGATAAACCACCATATAACCCGTTATCTCTTTCTCAAGTTTTTATATTAGCAGTACTTGTAATAAGCAACATATCCACGTAACACCTCATGTTCAAAATAGTTCTCCATGCCAG	DDDDDHIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIHIIHIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIHIIIIHHIIIIIHIIIIIIIIIHIIHIIHIIIIHIIIIIIIIIIHIIIIIIIHIIIIHIHHIIIIIIIHIIHIIIIGHHIIIIHIIIHGHIIIIIIIIGIGIHDFHHHIIIGIIIIHIIIIIHIHHHIHAFHHIIHIHIHHGE?HIIFIEHF	AS:i:242	X1:i:1	X2:i:0	XJ:i:0	XL:i:2	XR:i:0
1:808764-M1	0	NZ_JRYM01000009	92219	255	250S	*	0	0	TTCCCCGCGCCAGCGGGGATAAACCGAATAGCAATAGTCCATAGATTTGCGAAAACAGGTGTTCCCCGCGCCAGCGGGGATAAACCGGAGCCTGACGAGACTACTGAGGCCGTTCTGTCGAGTTCCCCGCGCCAGCGGGGATAAACCACCATATAACCCGTTATCTCTTTCTCAAGTTTTTATATTAGCAGTACTTGTAATAAGCAACATATCCACGTAACACCTCATGTTCAAAATAGTTCTCCATGCC	DDDDDHIIHHIHHIHIIIIIIIIIHIIHIHIGIIGIIHIIIIIIIIIIIIIIIIIIIIIIIIIIIIIHHHIIHIIIGHHGHHHHHIGHEHIIIIGIIHIHIFHIIHIIIHIIHHHIIIIIIIIIGHHHIIIIIHHHHCHHHIIHGIHIGHHIIIIIIHIIHIGGFHHHHIIGHHHHEHHHIIIHIIHHIE@GHHIIHIIIIGIHIIIHHHHIIEHIHIFEGHHC?EHIIIIHIIHIFHHHHGHIGFHHGH	AS:i:244	X1:i:1	X2:i:0	XJ:i:0	XL:i:2	XR:i:0
1:967780-M2	0	NZ_JRYM01000009	92219	255	235S	*	0	0	AAAAACCGGGCAATCGCAAAAAGGCGTAATGTGTTCCCCGCGCCAGCGGGGATAAACCTGTGTTTGCGGCATTAACGCTCACCAGCATTTCGTGTTCCCCGCGCCAGCGGGGATAAACCGACGTGGTCATGGGTGCTGCTGTTGCAGAGCCAGTGTTCCCCGCGCCAGCGGGGATAAACCGAGCAGATACACGGCTTTGTATTCCGTGCGCCCGTGTTCCCCGCGCCAGCGGGGA	DDDDDIHIIHHIIIHIHIHIIHIHHHHIGIHGHHIIIIIHHHIIIIHIIIIIIIIHIFHHHHHHHHHIHIDHIIEHIHHIIIIGEGHIIIGFHFIIIHHGIHIHIGIIIHIICGEHHGHHHHHHIHDHHHIIHHIGHHHIGHI?HHICEFEHIIGIIIIHHCHHIIIIIIIIHCEHH:FEHDDDDFFGHHHEHIHIHBGHF@G@HHH?=??HH?EH?GCEEC<C?CEHHEDDCD?	AS:i:235	X1:i:1	X2:i:0	XJ:i:2	XL:i:0	XR:i:1
1:1468852-M2	0	NZ_JRYM01000009	92219	255	250S	*	0	0	CCAGCGGGGATAAACCGGCAAAAACCGGGCAATCGCAAAAAGGCGTAATGTGTTCCCCGCGCCAGCGGGGATAAACCTGTGTTTGCGGCATTAACGCTCACCAGCATTTCGTGTTCCCCGCGCCAGCGGGGATAAACCGACGTGGTCATGGGTGCTGCTGTTGCAGAGCCAGTGTTCCCCGCGCCAGCGGGGATAAACCGAGCAGATACACGGCTTTGTATTCCGTGCGCCCGTGTTCCCCGCGCCAGCG	DDDDDIIIHIHIIIIIIIIIIIIIIIHDHIIIIIIIIIIIIIIGIIIGHIIIIIIIIIIIIIIIIIIIHIIHIIIIIGIIIIIIIIHIIHIHHHIIIHHIIHEHIIIIIIIIHIIHIIIIIIIIIIIIIIIIIIIIIIHIHIIIIIIIIIIIIIIHIIIIIIIIIIICCHHHHHHGEGEHIIIIIIGIIIGHH=HFHEGHHDDHIIIIGIIIIIDHGIHHIIHHCH.EHHHCHHIHHIIIGHII?>DHH?	AS:i:250	X1:i:1	X2:i:0	XJ:i:2	XL:i:0	XR:i:1
1:1858859-M1	0	NZ_JRYM01000009	92219	255	248S	*	0	0	GCGCCAGCGGGGATAAACCGAATAGCAATAGTCCATAGATTTGCGAAAACAGGTGTTCCCCGCGCCAGCGGGGATAAACCGGAGCCTGACGAGACTACTGAGGCCGTTCTGTCGAGTTCCCCGCGCCAGCGGGGATAAACCACCATATAACCCGTTATCTCTTTCTCAAGTTTTTATATTAGCAGTACTTGTAATAAGCAACATATCCACGTAACACCTCATGTTCAAAATAGTTCTCCATGCCAGAG	BADDAF?GHIIIHIIIIIIIGDIIIIIIIIIIIGIHIHCEHIHHHIHGHHHIHIIIIIHIHIIIIIHIIIIIHEGHHIIHIIIHIIGIHHGHHHHHHIIHHF1GHDHHEEHHHGEHHHHHIIHGIGCHHHGHICDHIHHHHGHIHIDGHHHEEHEHHHHIIHHIG0DGHGIHHIIIFHHHHG?/DGHHIHIIEHEH??GE@B?FHHHEEHIIIHDGHHHGHHHGHHFEHFHHHHHCEEAGB@H-BCFH	AS:i:242	X1:i:1	X2:i:0	XJ:i:0	XL:i:4	XR:i:0
1:2136141-M1	0	NZ_JRYM01000009	92219	255	248S	*	0	0	TCCCCGCGCCAGCGGGGATAAACCGAATAGCAATAGTCCATAGATTTGCGAAAACAGGTGTTCCCCGCGCCAGCGGGGATAAACCGGAGCCTGACGAGACTACTGAGGCCGTTCTGTCGAGTCCCCCGCGCCAGCGGGGATAAACCACCATATAACCCGTTATCTCTTTCTCAAGTTTTTATATTAGCAGTACTTGTAATAAGCAACATATCCACGTAACACCTCATGTTCAAAATAGTTCTCCATGC	DDDDDHDHIIIIIGIHIHIIIHHHHIIIIIIIHIGHIIH@EHHIIIHHIIIIGIIIIGFHIIIIIIIIIIIIIIHHIHHIIHIGEHHIIIIHIIIIIIIII<FHIIIHGHIIIIIIHHHIII11=EHHHIGIGHHIHIIICHEHIHIIHHIGGHHHHHIHHHHHIIIGGHIGHIHFEHIIHIIIEHGHHIIIIGIIIIGIIIIGIIIIIGHHHIHIHII-BHCHIHIHHIHFEH?FHEGE?GHHIG@H	AS:i:238	X1:i:1	X2:i:0	XJ:i:0	XL:i:1	XR:i:0
```

**Comment by cjfields** [comment]
Just a note, I see this using an older version of breseq (v0.28.1) as well as v0.31.1

**Comment by jeffreybarrick** [comment]
Thanks -- I see the problem and will see about fixing it.

This is a bug happening when _breseq_ thinks a read spans a new junction and normally splits it into an M1 and M2 portion in the bam output to show how it is matched each place. Except in this case it is incorrectly splitting a read that actually doesn't span the junction and the consequence is an M1 that is fully aligned and the M2 doesn't match at all.

**Comment by jeffreybarrick** [comment]
Checked in a fix for this. It will be included in the v0.32.0 version release that should be up soon.

**Comment by cjfields** [comment]
Thanks @jeffreybarrick !
